src/NovelRT/Graphics/GraphicsResourceManager.cpp

```diff
@@ -48,22 +71,51 @@ namespace NovelRT::Graphics
     GraphicsResourceManager& GraphicsResourceManager::operator=(GraphicsResourceManager&& other) noexcept
     {
         _graphicsDevice = std::move(other._graphicsDevice);
-        _stagingBuffer = Utilities::Lazy<std::shared_ptr<GraphicsBuffer>>([&]() { return CreateStagingBuffer(); });
-
-        if (other._stagingBuffer.isCreated())
-        {
-            _stagingBuffer.reset(other._stagingBuffer.getActual());
-            other._stagingBuffer.reset();
-        }
-
+        _stagingBuffers = std::move(other._stagingBuffers);
+        _defaultBuffers = std::move(other._defaultBuffers);
         _vertexBuffers = std::move(other._vertexBuffers);
+        _indexBuffers = std::move(other._indexBuffers);
+        _constantBuffers = std::move(other._constantBuffers);
         _textures = std::move(other._textures);
         _stagingBufferSize = other._stagingBufferSize;
         other._stagingBufferSize = 0;
+        _contextIndex = other._contextIndex;
+        other._contextIndex = 0;
 
         return *this;
     }
 
+    void GraphicsResourceManager::PrepForFrameWithContextIndex(size_t newContextIndex)
+    {
+        size_t finalContextCount = _contextIndex < newContextIndex ? newContextIndex : _contextIndex;
+
+        size_t currentStagingBufferContextSize = _stagingBuffers.size();
+        while (currentStagingBufferContextSize <= finalContextCount)
+        {
+            _stagingBuffers.emplace_back(std::vector<std::shared_ptr<GraphicsBuffer>>{});
```

Comment:
Does this move or copy here?

Reply:
on MSVC internally this just calls `push_back` I believe, so whatever that's doing is what's happening here. I was mostly just trying to get it to compile and work as I want.

Reply:
It might be worth looking into this later, then, as a potential optimisation that could be made. Not super important for now though.